### PR TITLE
Add documentation for dataframe operator

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

In the past, we had a tutorial which illustrated how to use each of our operators/decorators:
https://github.com/astronomer/astro-sdk/blob/be6280df00ccff0d7a1c0dfb099b2065303dbe88/REFERENCE.md

closes: #593 

## What is the new behavior?

Have a reference page per operator/decorator similar to
https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/ecs.html#howto-operator-ecsoperator

In which we reference parts of an (automated tested) example DAG which illustrates the usage of that operator/decorator.

Many of these use cases already exist in our example DAGs - we should reference them.

## Does this introduce a breaking change?
No

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: docs/astro/sql/operators/dataframe.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+======================================
+dataframe operator
+======================================
+
+.. _dataframe_operator:
+
+When to use the ``dataframe`` operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. ``dataframe`` operator converts a SQL table into a dataframe. Users can give a python function that takes a dataframe as one of its inputs and run that python function. Once that function has completed, the result is accessible via the Taskflow API.
```
Comment (dimberman) [inline docs/astro/sql/operators/dataframe.rst:9]:
```suggestion
``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. When a ``Table`` object is passed into the``dataframe`` operator, the operator automatically converts the SQL table into a dataframe. Users can then give a python function that takes a dataframe as one of its inputs and run that python function against the rendered dataset. Once the python function has completed, the result is accessible via the Taskflow API.
```

File: docs/astro/sql/operators/dataframe.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+======================================
+dataframe operator
+======================================
+
+.. _dataframe_operator:
+
+When to use the ``dataframe`` operator
```
Comment (dimberman) [inline docs/astro/sql/operators/dataframe.rst:7]: There should be a mention in this section about how a user can use the `output_table` parameter to turn the resulting dataframe into a table.

**Review by dimberman — CHANGES_REQUESTED** [review]

File: docs/astro/sql/operators/dataframe.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
+======================================
+dataframe operator
+======================================
```
Comment (feluelle) [inline docs/astro/sql/operators/dataframe.rst:3]:
```suggestion
==================
dataframe operator
==================
```

File: docs/astro/sql/operators/dataframe.rst [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+.. _dataframe_operator:
+
+When to use the ``dataframe`` operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The ``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. When a ``Table`` object is passed into the``dataframe`` operator, the operator automatically converts the SQL table into a dataframe. Users can then give a python function that takes a dataframe as one of its inputs and run that python function against the rendered dataset. Once the python function has completed, the result is accessible via the Taskflow API.
```
Comment (feluelle) [inline docs/astro/sql/operators/dataframe.rst:9]:
```suggestion
The ``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. When a ``Table`` object is passed into the ``dataframe`` operator, the operator automatically converts the SQL table into a dataframe. Users can then give a python function that takes a dataframe as one of its inputs and run that python function against the rendered dataset. Once the python function has completed, the result is accessible via the Taskflow API.
```

File: docs/astro/sql/operators/dataframe.rst [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+Example:
+    The following example aggregate the data and returns dataframe.
```
Comment (feluelle) [inline docs/astro/sql/operators/dataframe.rst:12]:
```suggestion
    The following example aggregates the data and returns a dataframe.
```

**Review by feluelle — APPROVED** [review]
LGTM, but please ask for another approval as I am not as experienced yet with the sdk. :)

File: docs/astro/sql/operators/dataframe.rst [hunk 6]
```diff
@@ -0,0 +1,17 @@
+==================
```
Comment (utkarsharma2) [inline docs/astro/sql/operators/dataframe.rst:1]:
```suggestion

.. dataframe_operator:

==================
```

File: docs/astro/sql/operators/dataframe.rst (hunk 2 above)
Comment (sunank200) [reply docs/astro/sql/operators/dataframe.rst:7]: @dimberman `DataframeOperator` is inheriting from `DecoratedOperator` in the current [code](https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/operators/dataframe.py#L76). This doesn't have `output_table`.

I feel it should be inherited from `BaseSQLDecoratedOperator` as in [base_decorator](https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/operators/base_decorator.py#L15). 

WDYT?

File: docs/astro/sql/operators/dataframe.rst [hunk 7]
```diff
@@ -0,0 +1,20 @@
+
+.. dataframe_operator:
+
+==================
+dataframe operator
+==================
+
+.. _dataframe_operator:
```
Comment (kaxil) [inline docs/astro/sql/operators/dataframe.rst:8]: Do you intend to have this same line on L2?

File: docs/astro/sql/operators/dataframe.rst [hunk 8]
```diff
@@ -0,0 +1,20 @@
+
+.. dataframe_operator:
```
Comment (kaxil) [inline docs/astro/sql/operators/dataframe.rst:2]:
```suggestion
.. _dataframe_operator:
```

File: docs/astro/sql/operators/dataframe.rst (hunk 7 above)
Comment (kaxil) [reply docs/astro/sql/operators/dataframe.rst:8]: If so, we should remove this.

```suggestion
```

File: docs/astro/sql/operators/dataframe.rst (hunk 8 above)
Comment (sunank200) [reply docs/astro/sql/operators/dataframe.rst:1]: [Done](https://github.com/astronomer/astro-sdk/pull/631/commits/46ed25c9f06bf5a843281352718547b3dc5f47f2)

File: docs/astro/sql/operators/dataframe.rst (hunk 7 above)
Comment (sunank200) [reply docs/astro/sql/operators/dataframe.rst]: [Done](https://github.com/astronomer/astro-sdk/pull/631/commits/46ed25c9f06bf5a843281352718547b3dc5f47f2)

File: docs/astro/sql/operators/dataframe.rst (hunk 2 above)
Comment (sunank200) [reply docs/astro/sql/operators/dataframe.rst:7]: [Done](https://github.com/astronomer/astro-sdk/pull/631/commits/46ed25c9f06bf5a843281352718547b3dc5f47f2) the change @dimberman

**Review by pankajastro — APPROVED** [review]

**Review by dimberman — APPROVED** [review]

**Pull request merged**